File: src/main/java/nowcoder/S45_IsContinuous.java
package nowcoder;

import java.util.Arrays;

/**
 * @author LiuZhiguo
 * @date 2020/2/8 12:24
 * 判断扑克牌是否为顺子
 */
public class S45_IsContinuous {
    public boolean isContinuous(int[] numbers){
        if (numbers == null || numbers.length == 0){
            return false;
        }
        int zeroNum = 0;//记录王的个数
        int count = 0;//记录排序后相邻两个数之间的间隔
        //将数组排序
        Arrays.sort(numbers);
        for (int i=0;i<numbers.length;i++){
            if (numbers[i] == 0)
                zeroNum++;
            if (numbers[i] != 0)
                break;
        }
        for (int i=zeroNum+1;i<numbers.length;i++){
            if (numbers[i] == numbers[i-1] )
                return false;
            count += numbers[i] - numbers[i-1] -1;
        }
        if(zeroNum >= count)
            return true;
        else return false;
    }
    public static void main(String[] args){
        S45_IsContinuous s45 = new S45_IsContinuous();
        int[] array = {1, 3, 0, 0, 7, 5, 2};
        System.out.println(s45.isContinuous(array));
    }
}

File: src/main/Python/nowcoder/S45_IsContinuous.py
class IsContinuous:
    def isContinuous(self, numbers):
        if not numbers:
            return False
        numbers.sort()
        zeroNum = numbers.count(0)
        for i, v in enumerate(numbers[:-1]):
            if v != 0:
                if numbers[i+1] == v:
                    return False
                zeroNum = zeroNum - (numbers[i+1] - v) + 1
                if zeroNum < 0:
                    return False
        return True
if __name__ =='__main__':
    test = IsContinuous()
    array = [1, 3, 0, 0, 7, 5, 2]
    print(test.isContinuous(array))